ksql-engine/src/main/java/io/confluent/ksql/codegen/SqlToJavaVisitor.java

```diff
@@ -145,7 +145,9 @@ private String formatExpression(final Expression expression) {
         final StringLiteral node,
         final Void context
     ) {
-      return new Pair<>("\"" + node.getValue() + "\"", Schema.OPTIONAL_STRING_SCHEMA);
+      return new Pair<>(
+          "\"" + node.getValue().replace("\"", "\\\"") + "\"",
```

Comment:
I feel like at some point we need to unescape the values, especially if we are then passing the string literals into UDFs that expect certain semantics. If I have:
```sql
SELECT CONCAT(CONCAT('\"', colA), '\"'))
```
I expect `\"colA\"` as the result.

Reply:
I don't think we need to un-escape the values. We need to convert the sql string into a java.util.String, which should have the quotes escaped. That said, I think your example raises a valid issue - currently this patch will generate an invalid java string literal: "\\""

Reply:
Using StringEscapeUtils addresses this. I've updated accordingly and added your example as a test case.